pkg/plugin/aggregation/run.go

```diff
@@ -30,6 +30,10 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+const (
+	gracefulShutdownPeriod = 45
```

Comment:
Do we need the distinction of can clean up just be the default case for normal timeout conditions?

Reply:
A normal timeout shuts down the aggregation server. Instead of that, we want to clean up the plugins first and give them a chance to report the data back to the aggregation server. After some amount of time we can then clean up the aggregation server.

We could bake this logic into the normal shutdown path. This would remove the newly introduced `for{select{` block which might be a good thing but will increase the complexity of a timeout (likely we'd have to sleep for this grace period anyway to give the plugins a chance to respond to being shut down).

Reply:
I think it makes more sense to keep them separate. If we add all this logic to the end cleanup, we have to do something to pause to give the plugins time to send results to the aggregator. If we do something like a sleep instead of this kind of loop, we might miss errors that come back since we'd be blocking this thread on the sleep and not the select statement.